qemu/qemu.go

```diff
@@ -439,6 +475,14 @@ func (fsdev FSDevice) QemuParams(config *Config) []string {
 		deviceParams = append(deviceParams, fmt.Sprintf(",devno=%s", fsdev.DevNo))
 	}
 
+	if fsdev.DisableModern {
+		deviceParams = append(deviceParams, ",disable-modern=true")
+	} else {
+		deviceParams = append(deviceParams, ",disable-legacy=on")
```

Comment:
Using `=on` in one case and `=true` in another seems pointlessly inconsistent.

You also won't need this if using `confidential-guest=support=` since it automatically sets the device parameter defaults as necessary for an encrypted guest.